ocular/src/chain/client/authz.rs

```diff
@@ -115,18 +116,20 @@ impl ChainClient {
                 expiration: expiration_timestamp,
             }),
         };
+        let account = self.query_account(granter.id.as_ref().to_string()).await?;
+        let (account_number, sequence) = (account.account_number, account.sequence);
 
         // Build tx body.
         let msg_any = prost_types::Any {
             type_url: String::from("/cosmos.authz.v1beta1.MsgGrant"),
             value: msg.encode_to_vec(),
         };
-
         let tx_body = tx::Body::new(vec![msg_any], &tx_metadata.memo, tx_metadata.timeout_height);
 
         self.sign_and_send_msg(
-            granter.public_key,
-            granter.private_key,
+            granter,
+            account_number,
+            sequence,
```

Comment:
Nit: No need for intermediary vars here, we can just pass back account.account_number & account.sequence